src/unitxt/fusion.py

```diff
@@ -89,10 +89,13 @@ def fusion_generator(self, split) -> Generator:
         weights = copy.deepcopy(self.weights)
         iterators = [iter(origin()[split]) for origin in self.origins]
         total_examples = 0
+        random_generator = new_random_generator(sub_seed="weighted_fusion")
```

Comment:
```suggestion
        random_generator = new_random_generator(sub_seed="weighted_fusion_" + split)
```
I think its better to have different seeds for different splits

Reply:
fixed